packages/protocol/test-sol/unit/common/FeeHandlerSeller.t.sol

```diff
@@ -0,0 +1,127 @@
+import { UniswapFeeHandlerSeller } from "@celo-contracts/common/UniswapFeeHandlerSeller.sol";
+
+contract FeeHandlerSellerTest is Test, TestConstants {
+  // Actors
+  address RECEIVER_ADDRESS = actor("Arbitrary Receiver");
+  address NON_OWNER_ADDRESS = actor("Arbitrary Non-Owner");
+
+  // Contract instances
+  GoldTokenMock celoToken; // Using mock token to work around missing transfer precompile
+  FeeHandlerSeller mentoFeeHandlerSeller;
+  FeeHandlerSeller uniswapFeeHandlerSeller;
+
+  // Contract addresses
+  address MOCK_CELO_TOKEN_ADDRESS;
+
+  // Helper data structures
+  FeeHandlerSeller[] feeHandlerSellerInstances;
+
+  function setUp() public {
+    celoToken = new GoldTokenMock();
+    MOCK_CELO_TOKEN_ADDRESS = address(celoToken);
+
+    mentoFeeHandlerSeller = new MentoFeeHandlerSeller(true);
+    uniswapFeeHandlerSeller = new UniswapFeeHandlerSeller(true);
+
+    feeHandlerSellerInstances.push(mentoFeeHandlerSeller);
+    feeHandlerSellerInstances.push(uniswapFeeHandlerSeller);
+  }
+}
+
+contract FeeHandlerSellerTest_Transfer is FeeHandlerSellerTest {
+  uint256 constant ZERO_CELOTOKEN = 0;
+  uint256 constant ONE_CELOTOKEN = 1000000000000000000;
+
+  function test_FeeHandlerSeller_ShouldTransfer_WhenCalledByOwner() public {
+    for (uint256 i = 0; i < feeHandlerSellerInstances.length; i++) {
+      celoToken.setBalanceOf(RECEIVER_ADDRESS, ZERO_CELOTOKEN); // Reset balance of receiver
+      assertEq(
+        celoToken.balanceOf(RECEIVER_ADDRESS),
+        ZERO_CELOTOKEN,
+        "Balance of receiver should be 0 at start"
+      );
+      celoToken.setBalanceOf(address(feeHandlerSellerInstances[i]), ONE_CELOTOKEN); // Faucet contract
+      assertEq(
+        celoToken.balanceOf(address(feeHandlerSellerInstances[i])),
+        ONE_CELOTOKEN,
+        "Balance of contract should be 1 at start"
+      );
+
+      vm.prank(feeHandlerSellerInstances[i].owner());
+      feeHandlerSellerInstances[i].transfer(
+        MOCK_CELO_TOKEN_ADDRESS,
+        ONE_CELOTOKEN,
+        RECEIVER_ADDRESS
+      );
+
+      assertEq(
+        celoToken.balanceOf(RECEIVER_ADDRESS),
+        ONE_CELOTOKEN,
+        "Balance of receiver should be 1 after transfer"
+      );
+      assertEq(
+        celoToken.balanceOf(address(feeHandlerSellerInstances[i])),
+        ZERO_CELOTOKEN,
+        "Balance of contract should be 0 after transfer"
+      );
+    }
+  }
+
+  function test_FeeHandlerSeller_ShouldRevert_WhenCalledByNonOwner() public {
+    for (uint256 i = 0; i < feeHandlerSellerInstances.length; i++) {
+      vm.prank(NON_OWNER_ADDRESS);
+
+      vm.expectRevert("Ownable: caller is not the owner");
+      feeHandlerSellerInstances[i].transfer(
+        MOCK_CELO_TOKEN_ADDRESS,
+        ONE_CELOTOKEN,
+        RECEIVER_ADDRESS
+      );
+    }
+  }
+}
+
+contract FeeHandlerSellerTest_SetMinimumReports is FeeHandlerSellerTest {
+  address ARBITRARY_TOKEN_ADDRESS = MOCK_CELO_TOKEN_ADDRESS;
```

Comment:
nit: would be good to get a random address that's different than the Celo one, just in case

Reply:
Great point, that's done in https://github.com/celo-org/celo-monorepo/pull/11155/commits/3af706f5483bb3a2ca01d939c668f8b272b0f395